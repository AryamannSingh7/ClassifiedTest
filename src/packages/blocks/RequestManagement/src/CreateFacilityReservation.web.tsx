import React from "react";
//components
import {
  Box,
  Button,
  Link,
  Typography,
  IconButton,
  Grid,
  TextareaAutosize,
  InputAdornment,
  TextField
} from "@material-ui/core";


//resources
import MailOutlineIcon from '@material-ui/icons/MailOutline';
import KeyboardBackspaceIcon from '@material-ui/icons/KeyboardBackspace';

import { Formik, Form, Field, ErrorMessage } from "formik";
import { withRouter } from 'react-router';
import Loader from "../../../components/src/Loader.web";
import { Input } from "react-native-elements";
import * as Yup from "yup";
import CountryCodeSelector from "../../country-code-selector/src/CountryCodeSelector";
import FacilityReservationController, { Props } from "./FacilityReservationController.web";
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import DateRangeOutlinedIcon from "@material-ui/icons/DateRangeOutlined";
//Customizable Area End

//resorces
import {
  Tenant_Logo,
  // Upload_Icon,
  // Clipboard_Icon,
  // Warning_Icon,
  House_Icon,
  // Box_Icon,
  Building1,
  LEADING_ICON,
  // Checkmark_Icon,
  // Error_Icon,
  TimeIcon
} from "../src/assets";
import { Visibility, VisibilityOff } from "@material-ui/icons";
import LockOpenIcon from '@material-ui/icons/LockOpen';
import moment from "moment";

class CreateFacilityReservation extends FacilityReservationController {
  constructor(props: Props) {
    super(props);
  }
  componentDidMount(): any {
    //@ts-ignore
    const id = this.props.history?.location?.id;
    if (id) {
      this.getFacilityReservationDetailsById(id);
    }
    //  else {
    //   this.props.history.push("/FacilityReservationDetails")
    //  }
    this.getMyApartmentList();
    // this.getIncidentRelated();
  }
  render() {
    const { navigation } = this.props;
    const id = this.state?.getFacilityReservationDetails?.id;
    const attributes = this.state?.getFacilityReservationDetails?.attributes;
    // console.log("commonAreaData=============>", this.state?.commonAreaData)
    console.log("attributes?.date =============>", attributes?.start_time, attributes?.end_time)
    console.log("moment(attributes?.date,'DD-MMM-YYYY').format('YYYY-MM-DD')=============>", moment(attributes?.date, 'DD-MMM-YYYY').format('YYYY-MM-DD'))
    return (
      <>
        <Box className="login-wrapper incident-wrapper">
          <Grid container spacing={2} className="auth-container">
            <Grid item xs={12} md={7} className="auth-cols">
              <Box className="common_content_block content-block">
                <Box className="content-header">
                  <Box className="left-block blocks">
                    <Box className="backIcons" onClick={() => window.history.back()}><KeyboardBackspaceIcon /></Box>
                    <h4>Update Facility Reservation</h4>
                  </Box>
                </Box>
                <Box className="content-block-wrapper common-incident-block desktop-ui create-reservation-wrapper">
                  <Formik
                    initialValues={{
                      areaReserve: attributes?.common_area?.id || " ",
                      buildingName: attributes?.building?.id || " ",
                      date: moment(attributes?.date, 'DD-MMM-YYYY').format('YYYY-MM-DD') || "",
                      timeFrom: attributes?.start_time || "",
                      timeTo: attributes?.end_time || "",
                      id: id || ""
                    }}
                    enableReinitialize
                    validationSchema={this.CreateFacilityReservationSchema()}
                    validateOnMount={true}
                    onSubmit={(values) => {
                      id ? (
                        this.updateFacilityReservation(values)
                      ) :
                        (
                          this.CreateFacilityReservation(values)
                        )
                    }}
                  >
                    {({ values, touched, errors, isValid, setFieldError, setFieldValue, handleChange }) => (
                      <Form translate="yes" className="commonForm CreateClassifiedFrm">
                        <h5 className="frm-title incident-preview-title"></h5>
                        <Box className="formGroup customSelect">
                          <FormControl variant="outlined" >
                            <span className="frmLeftIcons">
                              <img src={LEADING_ICON} className="frm-icons" alt="House Icon" />
                            </span>
                            <Select
                              name="buildingName"
                              labelId="demo-simple-select-outlined-label"
                              id="demo-simple-select-outlined"
                              style={{ paddingLeft: 50, marginTop: -3 }}
                              onChange={(e) => {
                                (e.target.value != " ") && setFieldValue("buildingName", e.target.value)
                              }}
                              value={values.buildingName}
                            >
                              <MenuItem disabled value=" ">
                                Building Name
                              </MenuItem>
                              {
                                this.state?.myApartmentList?.map((val: any, index: any) => (
                                  <MenuItem
                                    key={index}
                                    value={val?.id}
                                  >
                                    {`${val?.name}`}
                                  </MenuItem>
                                ))
                              }
                            </Select>
                            <ErrorMessage className="text-error" component="Typography" name="buildingName" />
                          </FormControl>
                        </Box>
                        <Box className="formGroup customSelect">
                          <FormControl variant="outlined" >
                            <span className="frmLeftIcons">
                              <img src={House_Icon} className="frm-icons" alt="House Icon" />
                            </span>
                            <Select
                              name="areaReserve"
                              labelId="demo-simple-select-outlined-label"
                              id="demo-simple-select-outlined"
                              style={{ paddingLeft: 50 }}
                              onChange={(e) => {
                                (e.target.value != " ") && setFieldValue("areaReserve", e.target.value)
                              }}
                              value={values.areaReserve}
                            >
                              <MenuItem disabled value=" ">
                                Area to Reserve
                              </MenuItem>
                              {
                                this.state?.commonAreaData?.map((val: any, index: any) => (
                                  <MenuItem
                                    key={index}
                                    value={val?.id}
                                  >
                                    {val?.attributes?.name}
                                  </MenuItem>
                                ))
                              }
                            </Select>
                            <ErrorMessage className="text-error" component="Typography" name="areaReserve" />
                          </FormControl>
                        </Box>
                        <Box className="DateSection">
                          <Grid container>
                            <Grid xs={12}>
                              <Box className="formGroup classifiedFormGroup">
                                <TextField
                                  label="date" variant="outlined"
                                  style={{ width: "100%", borderRadius: "25px", border: "1px solid #e9dede" }}
                                  type="date" name="date" fullWidth
                                  id="SurveyQuestion"
                                  format='DD/MM/YYYY'
                                  value={values?.date}
                                  onChange={handleChange}
                                  InputProps={{
                                    // min: "2019-01-24",
                                    //@ts-ignore
                                    max: "5000-05-31",
                                    startAdornment: (
                                      <InputAdornment position="start">
                                        {/* <DateRangeOutlinedIcon /> */}
                                      </InputAdornment>
                                    ),
                                    endAdornment: (
                                      <InputAdornment position="end">
                                        {/* <DateRangeOutlinedIcon /> */}
                                      </InputAdornment>
                                    ),
                                  }
                                  }
                                />
                                <ErrorMessage className="text-error" component="Typography" name="startDate" />
                              </Box>
                            </Grid>
                          </Grid>
                        </Box >
                        <Grid container>
                          <Grid xs={6} >
                            <Box className="formGroup classifiedFormGroup" >
                              <Field name="timeFrom" type="time" placeholder="From" className="formInput" format="hh:mm" />
                              <span className="frmLeftIcons">
                                <img src={TimeIcon} className="frm-icons" alt="Warning Icon" />
                              </span>
                              <ErrorMessage className="text-error" component="Typography" name="timeFrom" />
                            </Box>
                          </Grid>
                          <Grid xs={6}>
                            <Box className="formGroup">
                              <Field name="timeTo" type="time" placeholder="To" className="formInput formInputBox" format="hh:mm" />
                              <span className="frmLeftIcons">
                                <img src={TimeIcon} className="frm-icons" alt="Warning Icon" />
                              </span>
                              <ErrorMessage className="text-error" component="Typography" name="timeTo" />
                            </Box>
                          </Grid>
                          <Box className="reservationDec">
                            <h5>Description:</h5>
                            <p>
                              You can use garden for kids party, family gathering, building event etc. You are not allowed to have meals in the garden.
                            </p>
                            <br></br> <p>You will be charged <span>SR 50 per hour</span> for garden facility.</p>
                          </Box>
                        </Grid>
                        <Box className="customButton">
                          <Button variant="contained" type="submit">update request</Button>
                        </Box>
                      </Form>
                    )}
                  </Formik>
                </Box>

              </Box>
            </Grid>
            <Grid item xs={12} md={5} className="auth-cols">
              <Box className="right-block" display={{ xs: 'none', md: 'flex' }}>
                <img src={Building1.default} className="building-logo" alt="" />
              </Box>
            </Grid>
          </Grid>
        </Box>
        <Loader loading={this.state.loading} />
      </>
    )
  }
}

export default withRouter(CreateFacilityReservation)
