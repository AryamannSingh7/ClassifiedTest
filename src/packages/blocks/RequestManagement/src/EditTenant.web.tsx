import React from "react";
import {
  Button,
  Container,
  IconButton,
  withStyles,
  Box,
  Grid,
  MenuItem,
  Select,
  ListItemIcon,
  OutlinedInput,
  InputAdornment,
  Input,
  FormControl,
} from "@material-ui/core";
import { Link } from "react-router-dom";
import KeyboardBackspaceIcon from "@material-ui/icons/KeyboardBackspace";
import {
  BuildingImage,
  GreyTenantType,
  GreyTenantName,
  GreyEmailIcon,
  GreyBuildingName,
  GreyUnitNumber,
  GreyIdType,
  GreyIdNumber,
  GreyCalenderIcon,
  UploadIcon,
  GreyPhoneNumber,
  PdfIcon,
} from "./assets";
import { Formik, Form } from "formik";
import RegisterTenantController, { Props } from "./RegisterTenantController.web";
import { withTranslation } from "react-i18next";
import { TenantStyle } from "./TenantStyle.web";
import { CountryList } from "./countryList";
import CloseIcon from "@material-ui/icons/Close";
import Loader from "../../../components/src/Loader.web";

class EditTenant extends RegisterTenantController {
  constructor(props: Props) {
    super(props);
  }

  async componentDidMount(): Promise<void> {
    const tenant_id = this.props.navigation.getParam("id");
    this.setState({ tenantId: tenant_id }, () => {
      this.getTenantDetailsForEdit();
      this.getIdTypeList();
    });
  }

  render() {
    const { classes } = this.props;
    const { t }: any = this.props;

    console.log(this.state);

    return (
      <>
        <Loader loading={this.state.loading} />

        <Box style={{ background: "white", height: "100vh" }} className={classes.selectTemplate}>
          <Grid container>
            <Grid item xs={12} md={7}>
              <Box>
                <Box display={{ xs: "flex", md: "flex" }} className="top-bar">
                  <div className="left-icon">
                    <Link to={`/Tenant/${this.state.tenantId}`}>
                      <IconButton>
                        <KeyboardBackspaceIcon />
                      </IconButton>
                    </Link>
                    <span>{t("Edit A Tenant")}</span>
                  </div>
                </Box>
                <Container className="page-container">
                  <Box className="issue-lease-content">
                    <Formik
                      enableReinitialize
                      initialValues={this.state.registerTenantForm}
                      validationSchema={this.validationRegisterTenantFormSchema}
                      onSubmit={(values, { resetForm }) => {
                        this.handleEditTenant(values);
                      }}
                    >
                      {({ values, errors, touched, handleChange, handleBlur, handleSubmit, setFieldValue }) => {
                        return (
                          <Form onSubmit={handleSubmit} translate="true">
                            <Box className="select-input-box">
                              <FormControl fullWidth>
                                <Box className="select-box">
                                  <Select
                                    displayEmpty
                                    value={values.tenantType}
                                    onChange={(e: any) => {
                                      setFieldValue("tenantType", e.target.value);
                                    }}
                                    onBlur={handleBlur}
                                    name="tenantType"
                                    fullWidth
                                    className="select-input"
                                    input={<OutlinedInput />}
                                  >
                                    <MenuItem value="" disabled>
                                      {t("Type of Tenant")}
                                    </MenuItem>
                                    <MenuItem value="Individual">{t("Individual Person")}</MenuItem>
                                    <MenuItem value="Company" disabled>
                                      {t("Company")}
                                    </MenuItem>
                                  </Select>
                                  <img src={GreyTenantType} alt="" />
                                </Box>
                                {errors.tenantType && touched.tenantType && (
                                  <p className="error">{t(errors.tenantType)}</p>
                                )}
                              </FormControl>
                              <FormControl fullWidth>
                                <Input
                                  value={values.tenantName}
                                  onChange={(e: any) => {
                                    setFieldValue("tenantName", e.target.value);
                                  }}
                                  onBlur={handleBlur}
                                  name="tenantName"
                                  className="select-input input"
                                  placeholder={t("Tenant Name")}
                                  startAdornment={
                                    <InputAdornment position="start">
                                      <img src={GreyTenantName} alt="" />
                                    </InputAdornment>
                                  }
                                />
                                {errors.tenantName && touched.tenantName && (
                                  <p className="error">{t(errors.tenantName)}</p>
                                )}
                              </FormControl>
                              <FormControl fullWidth>
                                <Box className="mobile-box">
                                  <Select
                                    displayEmpty
                                    value={values.tenantCountryCode}
                                    onChange={(e: any) => {
                                      setFieldValue("tenantCountryCode", e.target.value);
                                    }}
                                    onBlur={handleBlur}
                                    name="tenantCountryCode"
                                    fullWidth
                                    className="mobile-select"
                                    input={<OutlinedInput />}
                                  >
                                    {CountryList.map((country: any) => {
                                      return (
                                        <MenuItem key={country.dial_code} value={country.dial_code}>
                                          <img
                                            src={`https://cdn.jsdelivr.net/npm/country-flag-emoji-json@2.0.0/dist/images/${
                                              country.code
                                            }.svg`}
                                            width="15"
                                            height="15"
                                            style={{ marginRight: "5px" }}
                                          />
                                          {country.dial_code}
                                        </MenuItem>
                                      );
                                    })}
                                  </Select>
                                  <Box className="divider" />
                                  <Input
                                    value={values.tenantMobile}
                                    onChange={(e: any) => {
                                      setFieldValue("tenantMobile", e.target.value);
                                    }}
                                    onBlur={handleBlur}
                                    name="tenantMobile"
                                    className="mobile-input"
                                    placeholder={t("Tenant Mobile")}
                                    startAdornment={
                                      <InputAdornment position="start">
                                        <img src={GreyPhoneNumber} alt="" />
                                      </InputAdornment>
                                    }
                                  />
                                </Box>
                                {errors.tenantMobile && touched.tenantMobile && (
                                  <p className="error">{t(errors.tenantMobile)}</p>
                                )}
                              </FormControl>
                              <FormControl fullWidth>
                                <Input
                                  value={values.tenantEmail}
                                  name="tenantEmail"
                                  className="select-input input"
                                  placeholder={t("Tenant Email ID")}
                                  startAdornment={
                                    <InputAdornment position="start">
                                      <img src={GreyEmailIcon} alt="" />
                                    </InputAdornment>
                                  }
                                  readOnly
                                />
                                {errors.tenantEmail && touched.tenantEmail && (
                                  <p className="error">{t(errors.tenantEmail)}</p>
                                )}
                              </FormControl>
                              <FormControl fullWidth>
                                <Input
                                  value={values.building}
                                  name="building"
                                  className="select-input input"
                                  startAdornment={
                                    <InputAdornment position="start">
                                      <img src={GreyBuildingName} alt="" />
                                    </InputAdornment>
                                  }
                                  readOnly
                                />
                              </FormControl>
                              <FormControl fullWidth>
                                <Input
                                  value={values.unit}
                                  name="unit"
                                  className="select-input input"
                                  startAdornment={
                                    <InputAdornment position="start">
                                      <img src={GreyUnitNumber} alt="" />
                                    </InputAdornment>
                                  }
                                  readOnly
                                />
                              </FormControl>
                              <FormControl fullWidth>
                                <Box className="select-box">
                                  <Select
                                    displayEmpty
                                    value={values.idType}
                                    onChange={(e: any) => {
                                      setFieldValue("idType", e.target.value);
                                    }}
                                    onBlur={handleBlur}
                                    name="idType"
                                    fullWidth
                                    className="select-input"
                                    input={<OutlinedInput />}
                                  >
                                    <MenuItem value="" disabled>
                                      {t("ID Type")}
                                    </MenuItem>
                                    {this.state.idTypeList.map((idType: any) => {
                                      return (
                                        <MenuItem value={idType.id} key={idType.id}>
                                          {idType.name}
                                        </MenuItem>
                                      );
                                    })}
                                  </Select>
                                  <img src={GreyIdType} alt="" />
                                </Box>
                                {errors.idType && touched.idType && <p className="error">{t(errors.idType)}</p>}
                              </FormControl>
                              <FormControl fullWidth>
                                <Input
                                  value={values.idNumber}
                                  onChange={(e: any) => {
                                    setFieldValue("idNumber", e.target.value);
                                  }}
                                  onBlur={handleBlur}
                                  name="idNumber"
                                  className="select-input input"
                                  placeholder={t("ID Number")}
                                  startAdornment={
                                    <InputAdornment position="start">
                                      <img src={GreyIdNumber} alt="" />
                                    </InputAdornment>
                                  }
                                />
                                {errors.idNumber && touched.idNumber && <p className="error">{t(errors.idNumber)}</p>}
                              </FormControl>
                              <FormControl fullWidth>
                                <Input
                                  value={values.idDate}
                                  onChange={(e: any) => {
                                    setFieldValue("idDate", e.target.value);
                                  }}
                                  onBlur={handleBlur}
                                  name="idDate"
                                  onFocus={(e: any) => (e.target.type = "date")}
                                  className="select-input input"
                                  placeholder={t("ID Expectation Date")}
                                  startAdornment={
                                    <InputAdornment position="start">
                                      <img src={GreyCalenderIcon} alt="" />
                                    </InputAdornment>
                                  }
                                />
                                {errors.idDate && touched.idDate && <p className="error">{t(errors.idDate)}</p>}
                              </FormControl>
                              <FormControl fullWidth>
                                <Box className="upload-box" onClick={() => this.uploadIDCard.click()}>
                                  <img src={UploadIcon} alt="" />
                                  <p>{t("Upload Tenant ID Card Copy")}</p>
                                </Box>
                                <input
                                  onChange={(e: any) => {
                                    setFieldValue("idCard", e.target.files);
                                  }}
                                  onBlur={handleBlur}
                                  name="idCard"
                                  style={{ display: "none" }}
                                  ref={(ref: any) => (this.uploadIDCard = ref)}
                                  accept=".pdf"
                                  type="file"
                                />
                                {[...values.idCard].map((file: any) => {
                                  return (
                                    <Box className="pdf-box">
                                      <img src={PdfIcon} alt="" />
                                      <Box className="pdf-info">
                                        <h4>{file.name}</h4>
                                        <CloseIcon
                                          onClick={() => {
                                            setFieldValue("idCard", []);
                                          }}
                                        />
                                      </Box>
                                    </Box>
                                  );
                                })}
                                {errors.idCard && touched.idCard && <p className="error">{t(errors.idCard)}</p>}
                              </FormControl>
                              <FormControl fullWidth>
                                <Box className="upload-box" onClick={() => this.uploadOtherDocument.click()}>
                                  <img src={UploadIcon} alt="" />
                                  <p>{t("Upload Other Tenant Documents")}</p>
                                </Box>
                                <input
                                  onChange={(e: any) => {
                                    const document: any = [...values.otherDocument, ...e.target.files];
                                    setFieldValue("otherDocument", document);
                                  }}
                                  onBlur={handleBlur}
                                  name="otherDocument"
                                  ref={(ref: any) => (this.uploadOtherDocument = ref)}
                                  accept=".pdf"
                                  type="file"
                                  style={{ display: "none" }}
                                  multiple
                                />
                                {[...values.otherDocument].map((file: any, index: number) => {
                                  return (
                                    <Box className="pdf-box">
                                      <img src={PdfIcon} alt="" />
                                      <Box className="pdf-info">
                                        <h4>{file.name}</h4>
                                        <CloseIcon
                                          onClick={() => {
                                            const remainFile = [...values.otherDocument].filter(
                                              (file: any, idx: number) => idx !== index
                                            );
                                            setFieldValue("otherDocument", remainFile);
                                          }}
                                        />
                                      </Box>
                                    </Box>
                                  );
                                })}
                                {errors.otherDocument && touched.otherDocument && (
                                  <p className="error">{t(errors.otherDocument)}</p>
                                )}
                              </FormControl>

                              <div className="next-button submit-button">
                                <Button type="submit">{t("Save")}</Button>
                              </div>
                            </Box>
                          </Form>
                        );
                      }}
                    </Formik>
                  </Box>
                </Container>
              </Box>
            </Grid>
            <Grid item xs={12} md={5}>
              <Box className="right-block right-image" display={{ xs: "none", md: "flex" }}>
                <img src={BuildingImage.default} className="building-logo" alt="" />
              </Box>
            </Grid>
          </Grid>
        </Box>
      </>
    );
  }
}

export default withTranslation()(withStyles(TenantStyle)(EditTenant));
