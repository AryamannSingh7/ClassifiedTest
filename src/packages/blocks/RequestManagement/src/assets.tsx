export const imgPasswordVisible = require("../assets/ic_password_visible.png");
export const imgPasswordInVisible = require("../assets/ic_password_invisible.png");
export const BuildingImage = require("../assets/building.png");
export const NoTenant = require("../assets/no-tenant.png");
export const EditIcon = require("../assets/edit.png");
export const DeleteIcon = require("../assets/delete.png");
export const TenantName = require("../assets/tenant-name.png");
export const TenantType = require("../assets/tenant-type.png");
export const BuildingName = require("../assets/building-name.png");
export const UnitNumber = require("../assets/unit-number.png");
export const CityIcon = require("../assets/city.png");
export const PhoneNumber = require("../assets/phone.png");
export const EmailIcon = require("../assets/email.png");
export const GreyTenantName = require("../assets/grey-tenant-name.png");
export const GreyTenantType = require("../assets/grey-tenant-type.png");
export const GreyBuildingName = require("../assets/grey-building-name.png");
export const GreyUnitNumber = require("../assets/grey-unit-number.png");
export const GreyPhoneNumber = require("../assets/grey-phone.png");
export const GreyEmailIcon = require("../assets/grey-email.png");
export const LeaseIcon = require("../assets/lease.png");
export const IdType = require("../assets/id-type.png");
export const GreyIdType = require("../assets/grey-id-type.png");
export const IdNumber = require("../assets/id-number.png");
export const GreyIdNumber = require("../assets/grey-id-number.png");
export const CalenderIcon = require("../assets/calender.png");
export const GreyCalenderIcon = require("../assets/grey-calender.png");
export const PdfIcon = require("../assets/pdf.png");
export const DownloadIcon = require("../assets/download.png");
export const UploadIcon = require("../assets/upload.png");

//FacilityReservation
export const Building1 = require("../assets/building1.png");
export const Tenant_Logo = require("../assets/TenantLogo.png");
export const Grid_Icon = require("../assets/grid-icon.png");
export const Filter_Icon = require("../assets/filter-icon.png");
export const Building_Logo = require("../assets/BuildingLogo.png");
export const Tick_Circle_Icon = require("../assets/tick-circle-icon.png");
export const TimeIcon = require("../assets/time_icon.png");
export const House_Icon = require("../assets/house-icon.svg");
export const LEADING_ICON= require("../assets/LEADING-ICON.png");
export const upcoming= require("../assets/upcoming.png");
export const pending= require("../assets/pending.png");
export const previous= require("../assets/previous.png");
export const Pencil= require("../assets/Pencil.png");
export const wrong= require("../assets/wrong.png");
export const Bin= require("../assets/Bin.png");
export const Close_Icon = require("../assets/icon-close.png");
export const Classified_CorrectIcon = require("../assets/classified_correct_icon.png");
export const User_Icon = require("../assets/user-icon.png");
export const Users_Icon = require("../assets/users-icon.png");