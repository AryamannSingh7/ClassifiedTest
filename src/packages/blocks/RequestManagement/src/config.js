Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.validationApiContentType = "application/json";
exports.validationApiMethodType = "GET";
exports.exampleAPiEndPoint = "EXAMPLE_URL";
exports.exampleAPiMethod = "POST";
exports.exampleApiContentType = "application/json";
exports.textInputPlaceHolder = "Enter Text";
exports.labelTitleText = "RequestManagement";
exports.labelBodyText = "RequestManagement Body";

exports.btnExampleTitle = "CLICK ME";

exports.ApiContentType = "application/json";

exports.apiMethodTypeGet = "GET";
exports.apiMethodTypePost = "POST";
exports.apiMethodTypeDelete = "DELETE";
// Customizable Area End