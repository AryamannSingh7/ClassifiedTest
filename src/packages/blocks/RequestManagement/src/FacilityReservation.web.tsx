// import React from "react";
import React, { useState } from 'react';
//components
import {
  Box,
  Button,
  Link,
  Typography,
  IconButton,
  Grid,
  Card,
  CardContent,
  CardActions,
  Menu,
  MenuItem
} from "@material-ui/core";

//resources
import MailOutlineIcon from '@material-ui/icons/MailOutline';
import KeyboardBackspaceIcon from '@material-ui/icons/KeyboardBackspace';

import { Formik, Form, Field } from "formik";
import { withRouter } from 'react-router';
import Loader from "../../../components/src/Loader.web";
import { Input } from "react-native-elements";
import * as Yup from "yup";
import CountryCodeSelector from "../../country-code-selector/src/CountryCodeSelector";
import FacilityReservationController, { Props } from "./FacilityReservationController.web";
//Customizable Area End
//resorces
import { Tenant_Logo, Building1, Grid_Icon, Filter_Icon } from "../src/assets";
class FacilityReservation extends FacilityReservationController {
  constructor(props: Props) {
    super(props);
  }
  //    componentDidMount():any {
  //      this.getIncidentListing(this.state.sortBy, this.state.status)
  //    }
  render() {
    const { navigation } = this.props;
    return (
      <>
        <Box className="login-wrapper incident-wrapper">
          <Grid container spacing={2} className="auth-container">
            <Grid item xs={12} md={7} className="auth-cols">
              <Box className="content-block">
                <Box className="content-header">
                  <Box className="left-block blocks">
                    <Box className="backIcons" onClick={() => window.history.back()}><KeyboardBackspaceIcon /></Box>
                    <h4>FacilityReservation</h4>
                  </Box>
                </Box>
                <Box className="content-block-wrapper facility-block-wrapper">
                  <Box className="incident-content-wrapper">
                    <Card className="incident-card card" >
                      <CardContent className="costom-card-content">
                        <Typography component="h4">
                          Upcoming Reservations
                        </Typography>
                        <CardActions className="card-footer">
                          {/* <Button className="success">Resolved</Button> */}
                          <Button variant="contained">{67}</Button>
                        </CardActions>
                      </CardContent>
                    </Card>
                  </Box>
                  <Box className="customButton add-incident">
                    <Button variant="contained" onClick={() => {
                      this.setState({ loading: true });//@ts-ignore
                      this.props.history.push("/CreateIncident")
                    }} >Add New Incident</Button>
                  </Box>
                </Box>
              </Box>
            </Grid>
            {/* desktop footer block */}
            <Grid item xs={12} md={5} className="auth-cols">
              <Box className="right-block" display={{ xs: 'none', md: 'flex' }}>
                <img src={Building1.default} className="building-logo" alt="" />
              </Box>
            </Grid>
          </Grid>
        </Box>
        <Loader loading={this.state.loading} />
      </>
    )
  }
}

export default withRouter(FacilityReservation)
