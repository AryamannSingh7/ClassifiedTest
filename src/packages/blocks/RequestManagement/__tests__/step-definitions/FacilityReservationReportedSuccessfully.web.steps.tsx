import { defineFeature, loadFeature } from "jest-cucumber";
import { mount } from "enzyme";
import React, { Component } from "react";
import { BrowserRouter as Router } from "react-router-dom"
import * as helpers from "../../../../framework/src/Helpers";
import FacilityReservation from "../../src/FacilityReservationReportedSuccessfully.web";
const navigation = require("react-navigation");

const screenProps = {
  navigation: {},
  id: "FacilityReservationReportedSuccessfully",
  location: jest.fn(),
  history: {
    push:jest.fn(),
  },
  match: jest.fn(),
  t:jest.fn()
};


const feature = loadFeature(
  "./__tests__/features/FacilityReservationReportedSuccessfully-scenario.web.feature"
);

jest.mock("react-i18next", () => ({
  withTranslation: () => (Component: any) => {
    Component.defaultProps = { ...Component.defaultProps, t: () => "" };
    return Component;
  },
}));

jest.mock("@material-ui/core/styles", () => ({
    withStyles: (styles: any) => (component: Component) => component,
  }));

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to FacilityReservationReportedSuccessfully", ({given,when,then}) => {
    let FacilityReservationMountWrapper: any;
    let instance: any;
    given("I am a User loading FacilityReservationReportedSuccessfully", () => {
      //@ts-ignore
        FacilityReservationMountWrapper = mount(<FacilityReservation.WrappedComponent {...screenProps} />,{ wrappingComponent: Router });
      });
  
      when("I navigate to the FacilityReservationReportedSuccessfully", () => {
        instance = FacilityReservationMountWrapper.instance();
      });
  
      then("FacilityReservationReportedSuccessfully will load with out errors", async () => {
        expect(FacilityReservationMountWrapper).toMatchSnapshot();
      });
    
    
  });
});
