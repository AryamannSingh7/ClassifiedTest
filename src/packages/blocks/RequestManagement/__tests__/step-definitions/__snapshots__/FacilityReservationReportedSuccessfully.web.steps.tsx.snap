// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilityReservationReportedSuccessfully User navigates to FacilityReservationReportedSuccessfully 1`] = `
<FacilityReservationReportedSuccessfully
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/FacilityReservationListing",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  id="FacilityReservationReportedSuccessfully"
  location={[MockFunction]}
  match={[MockFunction]}
  navigation={Object {}}
  t={[MockFunction]}
/>
`;
